### PR TITLE
[Audio metrics 02/12] Add Uni-VERSA regression with checkpoint-compatible parameters

### PR DESCRIPTION
**Pull request by ftshijt**
Add the Uni-VERSA regression predictor and its ESPnet model wrapper while preserving published checkpoint parameter names and shapes. This corrects loss direction, target indexing, optional-reference dimensions, and length-aware pooling without changing speaker-pooling implementations.

Both predictor layouts honor `loss_weights` supplied by metric name or legacy integer ID. Previously named weights raised `KeyError` in multi-branch training and were ignored in single-branch training. Default reductions remain unchanged; masked or zero-weight targets contribute no gradient.

Part 2/12 of the audio-metric series. The prerequisite #6747 has merged, and this branch is rebased onto upstream master `43904062365`. The Files changed tab contains only this component: **11 files, 954 added lines**.

Task/CLI registration is intentionally deferred to #6752 and #6753; this model-only PR remains independently testable.

Review fixes validate contiguous metric IDs and the reference-text vocabulary, return metric arrays of shape `(batch,)` in both predictor layouts, and align abstract annotations. Tests are split by source module and assert the exact training-stat keys. Parameter names and shapes remain unchanged.

Validation:

- **32 focused tests pass**, covering both branch layouts, pooling, normalization, masked targets, named/integer loss weights, frontend extraction, reference audio/text, missing references, non-mutating text padding, stats collection, and the legacy ignored `embedding_dim` setting.
- Local coverage of the added model code is **95%** (254 of 266 executable statements); this is not a claim that the hosted Codecov check has finished.
- **84 selected regression tests pass** with the dependent model/task/CLI/data code.
- The published `espnet/universa-wavlm_base_urgent24_multi-metric_noref` weights strictly load with all keys matched, complete a finite-gradient optimizer step with a verified parameter update, and run inference afterward.
- Black, isort, flake8, and whitespace checks pass. Local execution uses Python 3.10 / CPU PyTorch 2.11; supported-Python and hosted CI results remain pending.

Dependent drafts start at #6750. Reproducible published-checkpoint validation and compatibility limits for the complete series are documented in #6759.

CI follow-up (`f7cee6ee67b`): all six Python matrix failures and the Debian test failure stopped during collection because the new `test_espnet_model.py` collided with the enhancement test module. Adding `test/espnet2/universa/__init__.py` gives the tests distinct module names. The exact collision was reproduced locally before the fix; afterward the enhancement and UniVERSA suites passed together (327 passed, 48 skipped). With the dependent draft series, 342 passed and 48 skipped. Model code and checkpoint behavior are unchanged. Fresh hosted CI/Codecov results are pending.

Simplification (`804c67c7f59`): training and inference now share predictor computation and wrapper input preparation, removing 30 net lines across two files. Public signatures, module registration, autocast boundaries, and branch-specific loss reductions are preserved. Before/after checks matched state dictionaries, losses, statistics, gradients, and inference exactly across eight configurations. The combined suite passes 342 tests (48 skipped). Both published UniVERSA and ARECHO checkpoints pass strict loading, an optimizer update, and subsequent inference. Head construction and compatibility paths remain unchanged.

### CODE REVIEW

File: test/espnet2/universa/test_universa_base.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    assert set(result) == {"mos", "wer", "encoded_feat"}
+    assert all(np.isfinite(result[key]).all() for key in metrics)
+    # Padding must not affect a shorter utterance's predictions.
+    audio[2, 6:] = 1000
+    padded = model.inference(audio, lengths)
+    np.testing.assert_allclose(result["mos"], padded["mos"], atol=1e-5)
+
+
+@pytest.mark.parametrize("loss_fn", [masked_l1_loss, masked_mse_loss])
+def test_missing_labels_have_zero_loss_and_gradient(loss_fn):
```
[the remark this answers is not in the document]
Comment (ftshijt) [reply test/espnet2/universa/test_universa_base.py]: Addressed in ed95d4c13da. The masked-loss tests now live in `test_loss.py`, wrapper tests in `test_espnet_model.py`, and the shared model factory is a fixture in `conftest.py`. The focused suite passes all 32 tests with 96% local model-code coverage.

**Comment by ftshijt** [comment]
Thanks for the review. Follow-up to the [grouped review comments](https://github.com/espnet/espnet/pull/6749#pullrequestreview-5261547205), addressed in ed95d4c13da:

- Split tests by source module and moved the shared factory into a `conftest.py` fixture.
- Corrected the abstract `metrics` annotation to `Dict[str, torch.Tensor]` and the inference return annotation to include NumPy arrays and tensors.
- Made both predictor layouts return each metric with shape `(batch,)`; the projector test now covers both branch layouts.
- Added early validation for nonempty, unique, contiguous integer metric IDs and a positive vocabulary size when reference text is enabled, with regression tests for invalid configurations.
- Added exact assertions for the training-stat keys in both layouts.

Task registration is explicitly deferred to #6752, with CLI integration in #6753, to keep this model-only PR small and independently reviewable. The PR description now makes this dependency clear.

Validation: 32 focused tests passed with 96% local model-code coverage, and 84 selected tests passed across the combined series. The published UniVERSA checkpoint strictly loaded with all keys matched, completed a finite-gradient optimizer step with a verified parameter update, and ran inference afterward. Parameter names and shapes are unchanged.

Hosted CI is still pending; these local results are not a claim that all hosted checks have passed. The docstring-coverage warning in the bot summary is separate from executable test coverage and has not been addressed by this commit. The current Codecov comment references the previous commit (`f5dabbd`) and reports 60 missing uploads, so its aggregate project percentage is not a complete coverage comparison for the latest head.

**Review by sw005320 — APPROVED** [review]
LGTM!
